sql/sys_vars.cc

```diff
@@ -6467,3 +6468,83 @@ static Sys_var_charptr Sys_read_only_error_msg_extra(
     "which will be appended to read_only error messages.",
     GLOBAL_VAR(opt_read_only_error_msg_extra), CMD_LINE(OPT_ARG),
     IN_SYSTEM_CHARSET, DEFAULT(""), NO_MUTEX_GUARD, NOT_IN_BINLOG);
+
+#ifdef HAVE_SYS_GETTID
+class Change_priority : public Do_THD_Impl {
+ public:
+  Change_priority()
+      : m_changed(false), m_found(false), m_nice_val(0), m_thread_id(0) {}
+
+  virtual void operator()(THD *thd) {
+    if (!m_found && thd->thread_id() == m_thread_id) {
+      m_found = true;
+      const int ret =
+          setpriority(PRIO_PROCESS, thd->system_thread_id, m_nice_val);
+      if (ret != 0) {
+        my_error(ER_THRNICE_SETPRIORITY_FAILED, MYF(0), ret, strerror(errno));
+      } else {
+        // Update the variable used for resetting the nice value
+        thd->mark_thd_priority_as_alt();
+        m_changed = true;
+      }
+    }
+  }
+
+  bool parse_thread_nice_value(char *thd_id_nice_val) {
+    // parse input to get thread_id and nice_val
+    const std::string thd_id_nice_val_str(thd_id_nice_val);
+
+    const std::size_t delimpos = thd_id_nice_val_str.find(':');
+    if (delimpos == std::string::npos || delimpos == 0) return false;
```

Comment:
This if does not cover a case with trailing colon, like `42:`
may be
```suggestion
    if (delimpos == std::string::npos || delimpos == 0 || delimpos == thd_id_nice_val_str.size() - 1) return false;
```